lib/_http_outgoing.js

```diff
@@ -819,6 +830,13 @@ OutgoingMessage.prototype.end = function end(chunk, encoding, callback) {
     encoding = null;
   }
 
+  if (this.socket?.destroyed) {
```

Comment:
```suggestion
  if (this.destroyed) {
```

Reply:
`ERR_STREAM_WRITE_AFTER_END´ should have higher priority than `ERR_STREAM_DESTROYED`.